### PR TITLE
Implement Play Integrity attestation flow.

### PR DESCRIPTION
**Pull request by rosalyntan**
Add Play Integrity attestation logic, which includes obtaining a challenge nonce from the App Check backend.

### CODE REVIEW

File: appcheck/firebase-appcheck-playintegrity/src/main/java/com/google/firebase/appcheck/playintegrity/internal/GeneratePlayIntegrityChallengeRequest.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+ */
+public class GeneratePlayIntegrityChallengeRequest {
+
+  public GeneratePlayIntegrityChallengeRequest() {}
+
+  @NonNull
+  public String toJsonString() throws JSONException {
+    JSONObject jsonObject = new JSONObject();
+
+    return jsonObject.toString();
```
Comment (malcolmdeck) [inline appcheck/firebase-appcheck-playintegrity/src/main/java/com/google/firebase/appcheck/playintegrity/internal/GeneratePlayIntegrityChallengeRequest.java:33]: You just send an empty postBody? Can you please comment that that's intentional?

File: appcheck/firebase-appcheck-playintegrity/src/main/java/com/google/firebase/appcheck/playintegrity/internal/GeneratePlayIntegrityChallengeResponse.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  @VisibleForTesting static final String TIME_TO_LIVE_KEY = "ttl";
+
+  private String challenge;
+  private String timeToLive;
+
+  @NonNull
+  public static GeneratePlayIntegrityChallengeResponse fromJsonString(@NonNull String jsonString)
+      throws JSONException {
+    JSONObject jsonObject = new JSONObject(jsonString);
+    String challenge = emptyToNull(jsonObject.optString(CHALLENGE_KEY));
```
Comment (malcolmdeck) [inline appcheck/firebase-appcheck-playintegrity/src/main/java/com/google/firebase/appcheck/playintegrity/internal/GeneratePlayIntegrityChallengeResponse.java:41]: Nit: you may also consider #optString(key, null) if you prefer. No real difference, just whatever reads more cleanly to you

**Review by malcolmdeck — APPROVED** [review]
LGTM, Approval

File: appcheck/firebase-appcheck-playintegrity/src/main/java/com/google/firebase/appcheck/playintegrity/internal/GeneratePlayIntegrityChallengeRequest.java (hunk 1 above)
Comment (weixifan) [reply appcheck/firebase-appcheck-playintegrity/src/main/java/com/google/firebase/appcheck/playintegrity/internal/GeneratePlayIntegrityChallengeRequest.java:33]: Yes, this is correct -- the URL contains the app ID already.

**Review by weixifan — APPROVED** [review]
LGTM!

**Review by abzaloid — COMMENTED** [review]
the initial integration looks good, please also ping me when adding error handling, retry etc. thanks!!

File: appcheck/firebase-appcheck-playintegrity/src/main/java/com/google/firebase/appcheck/playintegrity/internal/GeneratePlayIntegrityChallengeRequest.java (hunk 1 above)
Comment (rosalyntan) [reply appcheck/firebase-appcheck-playintegrity/src/main/java/com/google/firebase/appcheck/playintegrity/internal/GeneratePlayIntegrityChallengeRequest.java:33]: Done.

File: appcheck/firebase-appcheck-playintegrity/src/main/java/com/google/firebase/appcheck/playintegrity/internal/GeneratePlayIntegrityChallengeResponse.java (hunk 2 above)
Comment (rosalyntan) [reply appcheck/firebase-appcheck-playintegrity/src/main/java/com/google/firebase/appcheck/playintegrity/internal/GeneratePlayIntegrityChallengeResponse.java]: Done.

**Pull request merged**